drgn_tools/kernfs_memcg.py

```
@@ -0,0 +1,224 @@
+    (to indicate current state of cgroup) and file cached by this page.
+
+    :params: max_pages: specify how many pages to find. By default first 10000 such
+    pages are listed. Use 0 to list all such pages.
+
+    """
+    PG_slab_mask = 1 << prog.constant("PG_slab")
+    mem_cgroup_root = prog["cgroup_subsys"][_MEMORY_CGRP_ID].root
+    total_count = 0
+    found_count = 0
+    for page in for_each_page(prog):
+        total_count = total_count + 1
+        try:
+            # Ignore slab pages
+            if page.flags & PG_slab_mask:
+                continue
+            # Ignore non page-cache pages
+            if not page.mapping:
+                continue
+            try:
+                mem_cgroup = page.mem_cgroup
+            except AttributeError:
+                mem_cgroup = page.memcg_data
+
+            if not mem_cgroup.value_() or mem_cgroup.value_() & 3:
+                continue
+            cgroup_subsys_state = cast(
+                "struct cgroup_subsys_state *", mem_cgroup
+            )
+            if cgroup_subsys_state.cgroup.root == mem_cgroup_root:
+                found_count = found_count + 1
+                cgrp = cgroup_subsys_state.cgroup
+                address_space = page.mapping
+                inode = address_space.host
+                if inode_path(inode) is None:
+                    continue
+                dentry = container_of(
+                    inode.i_dentry.first, "struct dentry", "d_u.d_alias"
+                )
+                path = dentry_path_any_mount(dentry).decode()
+                cgroup_state = decode_css_flags(cgrp.self.address_of_())
+                print(
+                    f"page: 0x{page.value_():x} cgroup: {cgroup_path(cgrp).decode()} state: {cgroup_state} path: {path}\n"
+                )
+                if max_pages and found_count == max_pages:
+                    break
+        except FaultError:
+            continue
```

I don't like silently swallowing errors, but I understand it can be necessary. Would you be willing to at least count the number of occurrences and log it at the end, so that we can understand whether there are actually any issues when the script runs?